imblearn/over_sampling/smote.py

```diff
@@ -59,12 +63,27 @@ class SMOTE(BaseOverSampler):
 
     kind : str, optional (default='regular')
         The type of SMOTE algorithm to use one of the following options:
-        ``'regular'``, ``'borderline1'``, ``'borderline2'``, ``'svm'``.
+        ``'regular'``, ``'borderline1'``, ``'borderline2'``, ``'svm'``,
+        ``'kmeans'``.
 
     svm_estimator : object, optional (default=SVC())
         If ``kind='svm'``, a parametrized :class:`sklearn.svm.SVC`
         classifier can be passed.
 
+    n_kmeans_clusters: int, optional (default=10)
```

Comment:
I think that we could make this a KMeans estimator instead.
I am thinking that it could be interesting to plug a MiniBatchKMeans for instance which could be quicker to converge and memory efficient. This could be the default. However, it let people to be able to pass a KMeans classifier if interested.

Reply:
Completely agree, MiniBatchKMeans is now the default and a test has been included to ensure that using a "normal" kmeans instance will also work.